src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -878,6 +914,7 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
         } else {
           List(javaLangObject())
         }
+      @unused val permits = permitsOpt()
       val (statics, body) = typeBody(INTERFACE)
       addCompanionObject(statics, atPos(pos) {
         ClassDef(mods | Flags.TRAIT | Flags.INTERFACE | Flags.ABSTRACT,
```

Comment:
```suggestion
        ClassDef(mods | Flags.TRAIT | Flags.INTERFACE | Flags.ABSTRACT | Flags.SEALED,
```

here? but see my other comment how this would affect exhaustivity checking.

Reply:
`mods` has already seen `sealed`, apparently.